### DISCUSSION TITLE
Queued process core dumping - debugging and next steps?

### DISCUSSION THREAD

**Discussion by j-schlesinger** [nig-sc/nigsc_homepage2#360] [Q&A] [answered]
I am trying to execute `sarek` from [nf-core](https://nf-co.re/) using the singularity backend, and the process seems to stop randomly and put a core.* file (or files) in the execution directory. I am uncertain how to proceed. I would like to debug this issue so that I can run nf-core jobs more reliably in the future.

I run `qsub` on this script:
```bash
#!/bin/bash

#$ -l medium
#$ -pe def_slot 4 
#$ -l s_vmem=32G
#$ -l mem_req=32G
#$ -l d_rt=10:00:00
#$ -l s_rt=10:00:00
#$ -cwd
#$ -V
#$ -v _JAVA_OPTIONS="-Xms2G -Xmx20G"
#$ -v NXF_OPTS='-Xms1G -Xmx4G'
#$ -v NXF_SINGULARITY_CACHEDIR=/lustre7/home/jschle/data/Sofl/illumina_hiseq_data/work/singularity
#$ -N Soph_Sarek_Nextflow_multicore
#$ -M jschlesi@nig.ac.jp
#$ -m a

./execute_sarek.sh
```
In addition, I have the environment variable `MALLOC_ARENA_MAX=3` in my local environment to overcome the JRE malloc issue.

Which runs `execute_sarek.sh` which looks like this:
```bash
#!/bin/bash

# Function to print log messages
RUNTIME=`date +%Y%m%dT%H%M%S`
_log() {
  local format='+%Y/%m/%d-%H:%M:%S'
  echo [`date $format`] "$@" >> sarek_log_$RUNTIME.log
}

_log Running Job $JOB_ID
ntfy -t "SC Execution Start" send "Execution started on SC job $JOB_ID"
nextflow run nf-core/sarek -profile singularity -params-file ./params.yaml &
CHILD=$!
_log Spawn nextflow with PID $CHILD
wait $CHILD
STATE=$?
ntfy -t "SC Execution End" send "Execution complete on SC job $JOB_ID with state $STATE"
_log finishing with $STATE
exit $STATE
```

Thank you for your assistance.

**Comment by sc-fs** [comment]
Please review your script.
・Please specify the interpreter used in the script.
・Since the environment variables are specified within the script,
          #$ -V
    is not necessary.
・The specified memory for the job is
      mem_req* def_slot= 128GB is being prepared.
    _JAVA_OPTIONS="-Xms2G -Xmx128G"
    I think that's fine.

```
#!/bin/bash

#$ -S /bin/bash
#$ -l medium
#$ -pe def_slot 4 
#$ -l s_vmem=32G
#$ -l mem_req=32G
#$ -l d_rt=10:00:00
#$ -l s_rt=10:00:00
#$ -cwd
#$ -v _JAVA_OPTIONS="-Xms2G -Xmx128G"
#$ -v NXF_OPTS='-Xms1G -Xmx4G'
#$ -v NXF_SINGULARITY_CACHEDIR=/lustre7/home/jschle/data/Sofl/illumina_hiseq_data/work/singularity
#$ -N Soph_Sarek_Nextflow_multicore
#$ -M jschlesi@nig.ac.jp
#$ -m a

./execute_sarek.sh
```

**Comment by j-schlesinger** [comment, marked as answer]
Thank you for your response! You pointed me in the right direction.

I explored the nf-core options and found that in addition to the `_JAVA_OPTIONS` and `NXF_OPTS` there is a job parameter of `max_memory`. It seems the workflow was running multiple instances of Java without knowing the memory limitation of the execution environment. I may still have issues, but at least for now the core dumping seems to have stopped.

For future reference to others:
- the max_memory and max_cpu parameters need to be set based on the grid engine environment.
- visit the [nf-core](https://nf-co.re/docs/usage/configuration#max-resources) documentation for more information.
- setting the process.executor configuration variable to 'sge' will allow nextflow to queue tasks directly, but you will want to either use `nohup` or other background task so that you won't need to stay connected for nextflow to run.
